src/functions/indAffineDirect.jl

```diff
@@ -0,0 +1,105 @@
+### CONCRETE TYPE: DIRECT PROX EVALUATION
+# prox! is computed using a QR factorization of A'.
+
+struct IndAffineDirect{R <: Real, T <: RealOrComplex{R}, M <: AbstractMatrix{T}, V <: AbstractVector{T}, F <: Factorization} <: IndAffine
```

Comment:
In fact all this could rely on caller (meaning user). It just would be fine to leave a note about it in the docs 😄  Just because julia errors about the type mismatch in this case are not very informative.

I solved my problem by extracting `T` from `A` and create zeros with this type.